ttnn/cpp/ttnn/operations/data_movement/pad/pad.hpp

```diff
@@ -0,0 +1,193 @@
+struct Pad {
+    static inline const std::array<TensorSchema, 1> input_tensor_schemas() {
+        return {ttnn::TensorSchema{
+            2,  // min rank
+            4,  // max rank
+            {ttnn::bfloat16, ttnn::bfloat8_b, ttnn::uint16, ttnn::int32, ttnn::uint32},
+            {ttnn::TILE_LAYOUT},
+            true,   // can_be_on_device
+            false,  // can_be_on_cpu
+            false,  // can_be_scalar
+            false   // is_optional}
+        }};
+    }
+
+    template <typename... Args>
+    static auto input_tensors_to_validate(const ttnn::Tensor& input_tensor, Args&&... args) {
+        return std::make_tuple(input_tensor);
+    }
+
+    static ttnn::Tensor execute_on_worker_thread(
+        uint8_t queue_id,
+        const ttnn::Tensor& input_tensor,
+        std::vector<uint32_t> output_padded_shape,
+        std::vector<uint32_t> input_tensor_start,
+        const float value,
+        const std::optional<MemoryConfig>& memory_config_arg) {
+
+
+        auto memory_config = memory_config_arg.value_or(input_tensor.memory_config());
+
+        auto output_tensor = operation::run(
+            tt::tt_metal::Pad{
+                .output_tensor_shape=tt::tt_metal::Shape(output_padded_shape),
+                .input_tensor_start=tt::tt_metal::Shape(input_tensor_start),
+                .pad_value=value,
+                .output_mem_config=memory_config,
+                .use_multicore=true
+            },
+            {input_tensor}).front();
+
+        return output_tensor;
+
+    }
+
+    static ttnn::Tensor execute_on_worker_thread(
+        uint8_t queue_id,
+        const ttnn::Tensor& input_tensor,
+        const Shape output_padded_shape,
+        const Shape input_tensor_start,
+        const float value,
+        const std::optional<MemoryConfig>& memory_config_arg) {
+
+
+        std::vector<uint32_t> output_padded_vector(output_padded_shape.rank());
+        std::vector<uint32_t> input_start_vector(output_padded_shape.rank());
+
+        for(uint32_t dim=0; dim<output_padded_shape.rank(); dim++) {
+            output_padded_vector[dim] = output_padded_shape[dim];
+            input_start_vector[dim] = input_tensor_start[dim];
+        }
+
+        return execute_on_worker_thread(queue_id, input_tensor, output_padded_vector, input_start_vector, value, memory_config_arg);
+    }
+
+
+    static ttnn::Tensor execute_on_worker_thread(
+        const ttnn::Tensor& input_tensor,
+        const Shape output_padded_shape,
+        const Shape input_tensor_start,
+        const float value,
+        const std::optional<MemoryConfig>& memory_config_arg) {
+
+
+
+        return execute_on_worker_thread(DefaultQueueId, input_tensor, output_padded_shape, input_tensor_start, value, memory_config_arg);
+
+    }
+
+
+
+    static ttnn::Tensor execute_on_worker_thread(
+        uint8_t queue_id,
+        const ttnn::Tensor& input_tensor,
+        std::vector<std::pair<uint32_t, uint32_t>> padding,
+        const float value,
```

Comment:
This API feels brittle to me.   The vector of pairs for padding is unbounded and has no constraint on the shape of the input tensor.    I see that we have TT_FATAL below, which will always run in a release.   It looks like a performance hit here.

Reply:
I do recognize the need for the reshape(s) is a far bigger perf hit.